lang/syn/src/idl/accounts.rs

```diff
@@ -192,7 +192,12 @@ fn get_pda(acc: &Field, accounts: &AccountsStruct) -> TokenStream {
     let parse_default = |expr: &syn::Expr| parse_seed(expr, accounts);
 
     // Seeds
-    let seed_constraints = acc.constraints.seeds.as_ref();
+    let seed_constraints = acc.constraints.seeds.as_ref().or_else(|| {
+        acc.constraints
+            .init
+            .as_ref()
+            .and_then(|init| init.seeds.as_ref())
+    });
```

Comment:
`constraints.seeds` and `constraints.init.seeds` are the same value:

https://github.com/solana-foundation/anchor/blob/ae1c158bee3f3a6eb2b01144d172b1d24056de82/lang/syn/src/parser/accounts/constraints.rs#L959
https://github.com/solana-foundation/anchor/blob/ae1c158bee3f3a6eb2b01144d172b1d24056de82/lang/syn/src/parser/accounts/constraints.rs#L1026

So the change doesn't seem to be doing anything other than achieving the same result in 6 lines rather than 1.

Reply:
@0x4ka5h The tests are not the same, see https://github.com/solana-foundation/anchor/issues/4057#issuecomment-3607423157